examples/kalman_filter.py

```diff
@@ -0,0 +1,374 @@
+        """Propagate state with constant velocity motion model on SE(2)."""
+        dt, linear_speed, angular_speed = self.split_input(sensor_input)
+        theta, x, y = state
+        x, y = state[1:] + dt * self.rotation_matrix(theta).dot(
+            linear_speed)
+        theta = theta + dt * angular_speed
+        theta = self.regularize_angle(theta)
+        return gs.concatenate((theta, [x, y]))
+
+    def propagation_jacobian(self, state, sensor_input):
+        """Construct the jacobian associated to the input.
+
+        Since the propagation writes f(x) = x*u, and the error is modeled on
+        the Lie algebra, the jacobian is Ad_{u^{-1}}.
+        """
+        dt, linear_speed, angular_speed = self.split_input(sensor_input)
+        input_vector_form = dt * gs.hstack((angular_speed, linear_speed))
+        input_inv = self.group.inverse(input_vector_form)
+
+        return self.adjoint_map(input_inv)
+
+    def noise_jacobian(self, state, sensor_input):
+        """Construct the jacobian associated to the process noise.
+
+        The noise being considered multiplicative, it is simply the identity
+        scaled by the time stamp.
+        """
+        dt, _, _ = self.split_input(sensor_input)
+        return gs.sqrt(dt) * gs.eye(self.dim_noise)
+
+    def observation_jacobian(self, state, observation):
+        """Construct the jacobian associated to the innovation."""
+        return gs.eye(2, 3, 1)
+
+    def get_measurement_noise_cov(self, state, observation_cov):
+        """Construct the measurement covariance for the innovation."""
+        theta, _, _ = state
+        rot = self.rotation_matrix(theta)
+        return rot.T.dot(observation_cov).dot(rot)
+
+    @staticmethod
+    def observation_model(state):
+        """Model used to obtain the measurements."""
+        return state[1:]
+
+    def innovation(self, state, observation):
+        """Discrepancy between the measurement and its expected value.
+
+        Here the linear error observation - expected is brought back to
+        the Lie algebra
+        """
+        theta, _, _ = state
+        rot = self.rotation_matrix(theta)
+        expected = Localization.observation_model(state)
+        return rot.T.dot(observation - expected)
+
+
+class KalmanFilter:
+    """Class for a general Kalman filter working on Lie groups.
+
+    Given an adapted model, it provides the tools to carry out non-linear state
+    estimation with an error modeled on the Lie algebra.
+    """
+
+    def __init__(self, model):
+        self.model = model
+        self.state = model.group.get_identity()
+        self.covariance = gs.zeros((self.model.dim, self.model.dim))
+        self.process_noise = gs.zeros(
+            (self.model.dim_noise, self.model.dim_noise))
+        self.measurement_noise = gs.zeros(
+            (self.model.dim_obs, self.model.dim_obs))
+
+    def initialise_covariances(self, prior_values, process_values, obs_values):
```

Comment:
initialize or init - geomstats has been using american English for the codebase so far.